app/controllers/v0/sessions_controller.rb

```diff
@@ -73,10 +78,14 @@ def parse_current_loa
       LOA::MAPPING[raw_loa]
     end
 
-    def create_new_user
-      user = User.new(user_attributes)
-      user = Decorators::MviUserDecorator.new(user).create unless user.loa1?
-      user
+    def saml_user
+      @user || create_saml_user
```

Comment:
It looks like it's for memoization. I'd be inclined to combine saml_user and create_saml_user like this:

```
def saml_user
  return @saml_user if defined?(@saml_user)
  @saml_user = User.new(user_attributes)
  @saml_user = Decorators::MviUserDecorator.new(@saml_user).create unless @saml_user.loa1?
  @saml_user
end
```

Both functions are small, so I'm not sure there's a need for a wrapper function just to handle the memoization.